src/System.Private.Xml/tests/Xslt/XsltCompiler/XsltCommon.cs

```diff
@@ -353,12 +354,31 @@ public static string SearchPath(string fileName)
             if (files.Count == 0)
                 throw new FileNotFoundException(fileName);
 
-            // Crudely prefer newer versions, eg 4.6.2 over 4.6.1,
-            // but it currently is not important
-            files.Sort(StringComparer.OrdinalIgnoreCase);
+            // Prefer newer versions, for stability
+            files.Sort((left, right) =>
+            {
+                int comparison = Comparer<float>.Default.Compare(GetVersion(left), GetVersion(right));
+
+                if (comparison == 0)
+                    comparison = string.Compare(left, right, StringComparison.OrdinalIgnoreCase);
+
+                return comparison;
+            });
 
             return files[files.Count - 1];
         }
+
+        // Pull the version out of a path like "C:\Program Files (x86)\Microsoft SDKs\Windows\v7.0A\bin\xsltc.exe"
+        private static float GetVersion(string s)
```

Comment:
Nit: make this a local function in SearchPath, or clarify the name to be clear about what it's getting a version from